scripts/commitcheck.sh

```diff
@@ -68,6 +69,11 @@ function check_tagged_line_with_url()
 function new_change_commit()
 {
     error=0
+    type_subject=$(git log -n 1 --pretty=%s "$REF" | cut -d: -f1)
+	# shellcheck disable=SC2076
+	[[ " ${types[*]} " =~ " ${type_subject} " ]] && error=0 || error=1; \
+	echo "error: ${type_subject} is not a valid subject type"
```

Comment:
Let's make this error message a little clearer, how about something like this which provides to list of known types.

```
error: the commit subject line must start with valid type, "${type_subject}" 
is not a recognized type. Valid types are: build, ci, docs, feat, fix, perf,
refactor, revert, style, coverity, and test.
```

Reply:
You are totally right, we shouldn't hide the new rules behind a vague error message!

Reply:
> Let's make this error message a little clearer, how about something like this which provides a list of known types.
> 
> ... Valid types are: build, ci, docs, feat, fix, perf,
> refactor, revert, style, coverity, and test.
> ```
Would be nice if the "types" variable from line 4 could be reused, so the valid types are definitly kept in sync in both places.

Reply:
@algragon Ofcourse :)